### ISSUE TITLE
Auto-type doesn't work with Citrix-RDP sessions

### ISSUE THREAD

**Issue by SeanCowburnMSFT** [keepassxreboot/keepassxc#3661]
[TIP]:  # ( Provide a general summary of the issue in the title above ^^ )
[TIP]:  # ( DO NOT include screenshots of your actual database! )
Auto-type does not seem to get the correct focus when changing to a Citrix-RDP session and the information is not passed on to the session. 

## Expected Behavior
[NOTE]: # ( Tell us what you expected to happen )
Auto-type should correctly gain focus of Citrix-RDP sessions, so that login information can be passed to apps running remotely.

## Current Behavior
[NOTE]: # ( Tell us what actually happens )
Keepassxc minimizes and seems to gain focus on the Citrix session, but no information is passed to the session. 

_Workaround:_ I increased the auto-type delay to 1 second and clicked into the username field in the app in the Citrix session, thereby actually gaining focus. This worked and both username and password were correctly entered.

On a side note: Copy & Paste also only works after actually clicking into the session, even though keepassxc minimizes and appears to change the focus to the Citrix session.

So I think it is safe to say that this is definitely an issue of keepassxc not gaining focus on/in the Citrix session correctly. (This works with keepass.)

## Possible Solution
[NOTE]: # ( Not required, but suggest a fix/reason for the bug )

## Steps to Reproduce
[NOTE]: # ( Provide a link to a live example, or an unambiguous set of steps to )
[NOTE]: # ( reproduce this bug. Include code to reproduce, if relevant )
1. Open a Citrix session and start an app within the session that requires login information (actually any text-editor will do - I also tested with notepad)
2. From keepassxc, select an entry and select auto-type.

## Context
[NOTE]: # ( How has this issue affected you? What unique circumstances do you have? )
Basically less comfort when using auto-type, which is a fantastic, time-saving feature.

## Debug Info
[NOTE]: # ( Paste debug info from Help → About here )
KeePassXC - Version 2.4.3
Revision: 5d6ef0c

Qt 5.12.3
Debugging mode is disabled.

Operating system: Windows 10 (10.0)
CPU architecture: x86_64
Kernel: winnt 10.0.18363

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare (signed and unsigned sharing)
- YubiKey

Cryptographic libraries:
 libgcrypt 1.8.4

**Comment by droidmonkey** [comment]
There is a difference between focusing on the local client window that *contains* the CITRIX session, and focusing on the app within that session. Ideally CITRIX would pass on the focus, but I guess it doesn't. I'm sure there is fancy windows code that can help us here, but I'm at a loss on where to start.

**Comment by SeanCowburnMSFT** [comment]
No doubt there is a difference and it is quite apparent in this use-case. As I said, KeePass is able to do this.

**Comment by Gryf1972** [comment]
It seems to be related to KeePassXC minimising itself when using auto-type. I can reproduce this without actually using auto-type: Two windows are open, one is the Citrix RDP session with a Notepad instance inside, and the other is KeePassXC (or any other program, e.g. Wordpad), which has the focus. When I now minimise KeePassXC via the window's button, the Citrix RDP window gets the focus, but even though the Notepad window inside the RDP session still has the focus, I still need to click into it before it actually accepts manual keystrokes.

This does not happen when I switch to the RDP window via Alt-Tab, nor when I send KeePassXC to the bottom (via VirtuaWin's "Push to Bottom" function) instead of minimising it.

**Comment by droidmonkey** [comment]
This request is better covered by https://github.com/keepassxreboot/keepassxc/issues/6548
